languages/python/oso/polar/query.py

```diff
@@ -105,24 +105,44 @@ def handle_external_call(self, data):
                     attr_typ = typ[attribute]
                     if isinstance(attr_typ, Relation):
                         rel = attr_typ
-                        # Use the fetcher for the other type to traverse the relationship
-                        build_query = self.host.types[rel.other_type].build_query
-                        exec_query = self.host.types[rel.other_type].exec_query
-                        assert build_query is not None
-                        assert exec_query is not None
-                        constraint = Filter(
-                            kind="Eq",
-                            field=rel.other_field,
-                            value=getattr(instance, rel.my_field),
-                        )
-                        constraints = [constraint]
-                        query = build_query(constraints)
-                        results = exec_query(query)
+                        adapter = self.host.adapter
```

Comment:
prob need to switch here too based on the same logic of whether we are using new / old data filtering